### PR TITLE
Fs export writer

### PR DESCRIPTION
**Pull request by anderjo4**
- the writer still needs more test-coverage.

### CODE REVIEW

**Comment by takesson** [comment]
- Naming: CmsExportWriterFsSingle
 - Use generic test files in this repo, tiny files.
 - There should be a reader as well.
 - Spelling: "Single" (Singel). "Preparing"
 - Soon a provider interface.

File: src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+	@Override
+	public void prepare(CmsImportJob job) {
+		logger.debug("Preparing reader");
+		
+		if (importJob != null) {
+			throw new IllegalStateException("Reader already consumed, initialize a new reader for each job.");
+		}
+		
+		if (!(job instanceof CmsImportJobSingle)) {
```
Comment (takesson) [inline src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java]: See CmsExportAwsWriterSingle.

File: src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		
+	}
+
+	private void createImportPath(CmsImportJob job) {
+		
+		Path fsParentPath = Paths.get(fsParent.getAbsolutePath());
+		StringBuilder sb = new StringBuilder();
+		sb.append(fsParentPath.toString());
+		
+		if (!fsParentPath.endsWith(File.separator)) {
```
Comment (takesson) [inline src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java]: Validate or normalize in constructor.

File: src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+			sb.append(File.separator);
+		}
+		sb.append(job.getJobPath());
+		
+		Path completeImportPath = Paths.get(sb.toString());
+		
+		if (!Files.exists(completeImportPath)) {
+			throw new IllegalStateException("Provided import path: " + completeImportPath + ", do not exist");
```
Comment (takesson) [inline src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java]: "Import path does not exist: " + completeImportPath

File: src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+		}
+		
+		if (!Files.isReadable(completeImportPath)) {
+			throw new IllegalStateException("Provided import path: " + completeImportPath + ", exists but is not readable.");
```
Comment (takesson) [inline src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java]: Similar to above.

File: src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+		}
+		
+		importPath = completeImportPath;
+	}
+
+	@Override
+	public Map<CmsExportMetaKey, String> getMeta() {
+		logger.warn("Reader do not support meta data returning null");
```
Comment (takesson) [inline src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java]: "Reader does not support metadata, returning null"

File: src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+		return null;
+	}
+
+	@Override
+	public InputStream getContents() {
+		
+		if (this.importPath == null) {
```
Comment (takesson) [inline src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java]: We should keep the paradigm of having a boolean. I generally dislike the this.importPath unless it is a necessary optimization.

File: src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java (hunk 1 above)
Comment (anderjo4) [reply src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java]: Do you mean CmsExportAwsReaderSingle?

File: src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+		
+		if (importJob != null) {
+			throw new IllegalStateException("Reader already consumed, initialize a new reader for each job.");
+		}
+		
+		if (!(job instanceof CmsImportJobSingle)) {
+			throw new IllegalArgumentException("Single readers expects CmsimportJobs that is of type CmsImportJobSingle");
+		}
+		
+		createImportPath(job);
```
Comment (takesson) [inline src/test/java/se/simonsoft/cms/item/export/CmsExportFsReaderSingle.java]: It is weird to place validation in this method.

**Pull request merged**